flows/events/run_expired.go

```diff
@@ -0,0 +1,34 @@
+package events
+
+import (
+	"github.com/nyaruka/goflow/flows"
+)
+
+// TypeRunExpired is the type of our flow expired event
+const TypeRunExpired string = "run_expired"
+
+// RunExpiredEvent events are sent by the caller to notify the engine that a run has expired
+//
+// ```
+//   {
+//     "type": "run_expired",
+//     "created_on": "2006-01-02T15:04:05Z",
+//     "run_uuid": "0e06f977-cbb7-475f-9d0b-a0c4aaec7f6a"
+//   }
+// ```
+//
+// @event run_expired
+type RunExpiredEvent struct {
+	BaseEvent
+	RunUUID flows.RunUUID `json:"run_uuid"    validate:"required,uuid4"`
+}
+
+// Type returns the type of this event
+func (e *RunExpiredEvent) Type() string { return TypeRunExpired }
+
+// Apply applies this event to the given run
+func (e *RunExpiredEvent) Apply(run flows.FlowRun) {
+	// TODO check this is correct run
```

Comment:
Oh yeah I meant to flag this for discussion because maybe we should aim for some consistency about what events can do in their `Apply` methods, and whether a check like "does this run exist in the session" belong in a validation stage rather than here.